test/e2e/service.go

```diff
@@ -112,25 +116,25 @@ var _ = Describe("Services", func() {
 		addEndpointPodOrFail(c, ns, name1, labels, []api.ContainerPort{{ContainerPort: 80}})
 		names = append(names, name1)
 
-		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name1: {80}})
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]string{name1: {portname1}})
```

Comment:
If endpoints for a pod do not inherit container IP and container port, we have to use something like the service port name to find the correct endpoints for the test. That's why I changed the port into the port name and adapted the test condition accordingly.